### ISSUE TITLE
Can't get Jint when no script tag is included

### ISSUE THREAD

**Issue by SirKurt** [AngleSharp/AngleSharp.Scripting#28]
I am trying to get Jint engine from the javascript service. However, when there is no script tag with a body, `GetJint` method returns null. I can get Jint engine after adding a script dynamically or including a script tag with body at start. It can be tested with following method.

```cs
[TestMethod]
public async Task JintTest()
{
    var cfg = Configuration.Default.WithJavaScript().WithDefaultLoader();
    var svc = cfg.Services.OfType<ScriptingService>().FirstOrDefault(x => x.Engine.Type == "text/javascript");

    var htmlEmpty = "<!doctype html><script type=\"text/javascript\"></script>";
    var htmlNonEmpty = "<!doctype html><script type=\"text/javascript\">a=5</script>";

    var document = await BrowsingContext.New(cfg).OpenAsync(m => m.Content(htmlNonEmpty));
    Assert.IsNotNull(svc.Engine.GetJint(document), "Can't get Jint when the script tag is not empty");

    document = await BrowsingContext.New(cfg).OpenAsync(m => m.Content(htmlEmpty));
    Assert.IsNotNull(svc.Engine.GetJint(document), "Can't get Jint when the script tag is empty");  // Currently error is here

    var script = document.CreateElement("script");
    script.SetAttribute("type","text/javascript");
    script.TextContent = "a=5";
    document.Head.AppendChild(script);

    Assert.IsNotNull(svc.Engine.GetJint(document), "Can't get Jint even after dynamically adding the script");
}
```

**Comment by FlorianRappl** [comment]
Hm the question is now if this is an AngleSharp problem or an AngleSharp.Scripting problem.

I think AngleSharp should support the notion of appending a script later on, where the script execution is triggered once the element is appended. So if the script is really executed then AngleSharp is behaving as it should be.

That being said Jint is only active for a document if it has been invoked for this document. Otherwise, no JavaScript engine is created. Potentially, the `GetJint` should be changed to `GetOrCreateJint`, but right now its simply a `GetJint` and results in `null` if no JavaScript has yet been executed.

One more thing to try in your code: As the script execution is asynchronous have you tried, e.g., inserting a `await Task.Delay(1000);` between appending the script element and getting the engine? 1 second is definitely over the top here, but if that is not working then no script is executed for sure.

**Comment by SirKurt** [comment]
I tried delaying at it still doesn't work. Note that I can get Jint after running a script on the page through a tag.

I considered the Jint is created lazily (for performance purposes maybe). The problem is I can't run a Javascript code before atleast one javascript code is run through `<script>` tag. So should I go out of the way to dynamically add a dummy script tag in case `GetJint` might return null?

Example counter case in browsers: Go to a [plain text link](https://wordpress.org/plugins/about/readme.txt) which makes sure no script is included. Open console (F12). Write your script in console and it works.

Of course that doesn't mean a browser includes the scripting engine in any case before you open the developer tools. I am just inquiring if the current `GetJint` works as intended.

**Comment by FlorianRappl** [comment]
The scenario you describe is still possible. How else what script elements evaluate their input? `GetJint` only gets an already created context; what you want is the `EvaluateScriptAsync` functionality. This is also the functionality that would be executed by a browser.

I don't know if the previously described behavior (adding a script tag does not add an engine) is a bug or not yet, but `GetJint` does what its asked for: Get Jint if its already available. It does not create one for you (otherwise it should be called `GetOrCreateJint` as mentioned).

Long story short: The `GetJint` works as designed.

**Comment by FlorianRappl** [comment]
Available in latest version.
